packages/bsky/src/migrate-moderation-data.ts

```diff
@@ -0,0 +1,280 @@
+  return
+}
+
+const setReportedAtTimestamp = async (db: PrimaryDatabase) => {
+  console.log('Initiating lastReportedAt timestamp sync')
+  const didUpdate = await sql`
+    UPDATE moderation_subject_status
+    SET "lastReportedAt" = reports."createdAt"
+    FROM (
+      select "subjectDid", "subjectUri", MAX("createdAt") as "createdAt"
+      from moderation_report
+      where "subjectUri" is null
+      group by "subjectDid", "subjectUri"
+    ) as reports
+    WHERE reports."subjectDid" = moderation_subject_status."did"
+      AND "recordPath"=''
+  `.execute(db.db)
+
+  console.log(
+    `Updated lastReportedAt for ${didUpdate.numUpdatedOrDeletedRows} did subject`,
+  )
+
+  const contentUpdate = await sql`
+    UPDATE moderation_subject_status
+    SET "lastReportedAt" = reports."createdAt"
+    FROM (
+      select "subjectDid", "subjectUri", MAX("createdAt") as "createdAt"
+      from moderation_report
+      where "subjectUri" is not null
+      group by "subjectDid", "subjectUri"
+    ) as reports
+    WHERE reports."subjectDid" = moderation_subject_status."did"
+      AND "recordPath" is not null
+      AND POSITION(moderation_subject_status."recordPath" IN reports."subjectUri") > 0
+  `.execute(db.db)
+
+  console.log(
+    `Updated lastReportedAt for ${contentUpdate.numUpdatedOrDeletedRows} subject with uri`,
+  )
+}
+
+const createStatusFromActions = async (db: PrimaryDatabase) => {
+  const allEvents = await db.db
+    // @ts-ignore
+    .selectFrom('moderation_action')
+    // @ts-ignore
+    .where('reversedAt', 'is', null)
+    // @ts-ignore
+    .select((eb) => eb.fn.count<number>('id').as('count'))
+    .executeTakeFirstOrThrow()
+
+  const chunkSize = 2500
+  const totalChunks = Math.ceil(allEvents.count / chunkSize)
+
+  console.log(`Processing ${allEvents.count} actions in ${totalChunks} chunks`)
+
+  await db.transaction(async (tx) => {
+    let currentChunk = 1
+    let lastProcessedId = 0
+    do {
+      const eventsQuery = tx.db
+        // @ts-ignore
+        .selectFrom('moderation_action')
+        // @ts-ignore
+        .where('reversedAt', 'is', null)
+        // @ts-ignore
+        .where('id', '>', lastProcessedId)
+        .limit(chunkSize)
+        .selectAll()
+      const events = (await eventsQuery.execute()) as ModerationActionRow[]
+
+      for (const event of events) {
+        // Remap action to event data type
+        const actionParts = event.action.split('#')
+        await adjustModerationSubjectStatus(tx, {
+          ...event,
+          action: `${actionParts[0]}#modEvent${actionParts[1]
+            .charAt(0)
+            .toUpperCase()}${actionParts[1].slice(
+            1,
+          )}` as ModerationEventRow['action'],
+          comment: event.reason,
+          meta: null,
+        })
+      }
+
+      console.log(`Processed events chunk ${currentChunk} of ${totalChunks}`)
+      lastProcessedId = events.at(-1)?.id ?? 0
+      currentChunk++
+    } while (currentChunk < totalChunks)
```

Comment:
We start at chunk `1` and remain less than `totalChunks`—at a glance it seems like this may lose one chunk.  For example, if there are two chunks then we might process chunk `1`, increment it to `2`, then stop before processing chunk `2` due to the `while ()` condition.

If this seems like we could get rid of chunk-counting altogether, and instead paginate through the result set as we're currently already doing until there are no results left.  The condition might look something like:
```suggestion
      lastProcessedId = events.at(-1)?.id
    } while (lastProcessedId !== undefined)
```